src/esp/gfx/MaterialData.h

```diff
@@ -35,19 +50,34 @@ struct PhongMaterialData : public MaterialData {
   ESP_SMART_POINTERS(PhongMaterialData)
 };
 
-struct PBRMaterialData : public MaterialData {
+struct PbrMaterialData : public MaterialData {
+  PbrMaterialData() : MaterialData(MaterialDataType::PBR){};
   Magnum::Color4 baseColor{0.7};
   Magnum::Float roughness = 0.9f;
   Magnum::Float metallic = 0.1f;
   Magnum::GL::Texture2D* baseColorTexture = nullptr;
   Magnum::GL::Texture2D* normalTexture = nullptr;
+
+  // Question:
+  // What if e.g., both metallicTexture and nonRoughnessMEtallicTexture exist?
+  // Answer:
+  // Such conflict will be handled in the PbrShader based on the priority of the
+  // textures. See generateCorrectFlags() in PbrShader for details.
+
+  // an independent, NOT packed texture to store "metallic" (in R
+  // channel)
   Magnum::GL::Texture2D* metallicTexture = nullptr;
+  // an independent, NOT packed texture to store "roughness" (in R channel)
   Magnum::GL::Texture2D* roughnessTexture = nullptr;
+  // a packed texture to store "roughness" (in G channel) and "metallic" (in B
+  // channel)
   Magnum::GL::Texture2D* noneRoughnessMetallicTexture = nullptr;
+  // a packed texture to store "occlusion"(in R channel), "roughness" (in G
+  // channel) and "metallic" (in B channel)
   Magnum::GL::Texture2D* occlusionRoughnessMetallicTexture = nullptr;
```

Comment:
This packing is a superset of the above, so you'd need maybe just one of the two?

Reply:
I do not think so.
In terms of design, I have different philosophy from the magnum library at this point.
- First, this is just a pointer, so the storage cost is tiny.
- It gives me a lot of convenience, avoiding any "if" control logic. For example, in Magnum, if roughnessTexture contains a texture. Later when you use it, you do not know if it is an independent texture, or a packed texture. You will have to call `hasOcclusionRoughnessMetallicTexture()` to tell which case it is. (In that function, there are "if clauses" for sure.) In my case, I do not need such things. If `occlusionRoughnessMetallicTexture` is not empty, it means it has a packed texture, and based on the priority among the textures, I do not even need to care what is in the `roughnessTexture`. In this manner, the code is simplified and cleaner.